mbed_build/_internal/templates/mbed_config.tmpl

```diff
@@ -1,28 +1,63 @@
-/*
-* Copyright (C) 2020 Arm Mbed. All rights reserved.
-* SPDX-License-Identifier: Apache-2.0
-*/
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
 
-// Automatically generated configuration file.
-// DO NOT EDIT, content will be overwritten.
+# Automatically generated configuration file.
+# DO NOT EDIT, content will be overwritten.
+add_library(gen_config INTERFACE)
+set(MBED_TOOLCHAIN "{{toolchain_name}}" CACHE STRING "")
+set(MBED_TARGET "{{target_name}}" CACHE STRING "")
+set(MBED_CPU_CORE "{{core}}" CACHE STRING "")
 
-#ifndef __MBED_CONFIG_H__
-#define __MBED_CONFIG_H__
+if(MBED_TOOLCHAIN STREQUAL "GCC_ARM")
+    add_definitions(-DTOOLCHAIN_GCC_ARM -DTOOLCHAIN_GCC)
+elseif(MBED_TOOLCHAIN STREQUAL "ARM")
+    add_definitions(-DTOOLCHAIN_ARM)
+endif()
 
-{% if options -%}// Configuration parameters {% endif %}
+set_property(GLOBAL APPEND PROPERTY MBED_TARGET_LABELS{% for label in labels %}
+    {{label}};
+{%- endfor %}
+)
+
+# target
+add_definitions({% for component in components %}
+    -DCOMPONENT_{{component.upper()}}=1
```

Comment:
Is there a decent way to expose these variables to CMake without `add_definitions`? It should be up to CMake files in Mbed OS to decide what to do with these variables.

Reply:
We could just set a variable, I suppose. Requires some more CMake work in mbed to do that.

Reply:
We're going to have to make some minor changes to the template when we do the "Use Modern CMake" story. Might be worth figuring out how best to expose these defines as part of that task.

Reply:
OK. Let's do it then.